File: python/1436B.py
# 1436B. Prime Squares 
# Completed in 140 ms 0 kb
# 03/15/2021

def prime_square(n):
    result = []
    result.append([1] + [0]*(n-2) + [1])
    index = 0

    for i in range(0, n-1):
        row = [0] * n
        row[index] = 1
        row[index+1] = 1
        result.append(row)
        index += 1

    return result

t = int(input())
for i in range(t):
    n = int(input())
    square = prime_square(n)
    for row in square:
        print(*row)

File: cpp/1436B.cpp
// 1436B. Prime Square
// Completed in 30 ms 0 kb
// 03/15/2021

#include <iostream>
#include <vector>
#include <algorithm>
#include <cmath>
#define ll long long
#define newl '\n'

using namespace std;

int main() {
    int t;
    cin >> t;
    while (t--) {
        int n;
        cin >> n;

        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                if (i == j || (i+1) % n == j) {
                    cout << 1 << " ";
                }
                else {
                    cout << 0 << " ";
                }
            }
            cout << newl;
        }
    
    }
    return 0;
}
